source/fundamentals/typescript.txt

```diff
@@ -70,90 +72,51 @@ You can pass a type parameter that extends the ``Document`` interface like this:
 Any Type
 ~~~~~~~~
 
-The following classes accept any type parameter:
+The following classes accept any type parameter that is not recursive:
 
 - `FindCursor <{+api+}/classes/FindCursor.html>`__
 - `AggregationCursor <{+api+}/classes/AggregationCursor.html>`__
 
+To view an example of a recursive type, see the :ref:`<node-driver-limitations>` section.
+
 You can find a code snippet that shows how to specify a type for the ``FindCursor``
 class in the
 :ref:`Find Multiple Documents Usage Example <node-driver-find-usage-example-code-snippet>`.
 
+.. _node-driver-limitations:
+
 Limitations
 -----------
 
-.. _node-driver-typescript-limitations-dot-notation:
-
-The driver cannot infer the type of values with keys containing **dot
-notation**. Dot notation is a property access syntax for navigating BSON objects.
-Click on the tabs to see code snippets that highlight this behavior:
-
-.. tabs::
-
-  .. tab:: Dot Notation
-     :tabid: dot-notation
-
-     The following code snippet does not raise a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-no-error
-        :end-before: end-no-error
-
-  .. tab:: Nested Objects
-     :tabid: nested-objects
-
-     The following code snippet raises a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
+You cannot specify a recursive type as a type parameter for {+driver-long+}
```

Comment:
Responding by paragraph:

- listed out the classes individually to try and clarify this
- I don't believe the suggested edit is accurate. It is not "classes that extend document", but rather some classes have type parameters that extend document.
- TODO: respond

Reply:
> I don't believe the suggested edit is accurate. It is not "classes that extend document", but rather some classes have type parameters that extend document.

Good to know -- why don't we workshop this together so that we can come up with a description that the reader and I can more quickly understand?